cli/src/manager/node/mod.rs

```diff
@@ -0,0 +1,75 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+mod error;
+
+use reqwest::Url;
+
+use crate::store::node::{FileBackedNodeStore, NodeStore};
+
+pub use crate::store::node::Node;
+pub use error::NodeManagerError;
+
+pub struct NodeManager {
```

Comment:
There is a lot of abstraction here, why do we need to wrap NodeStore in a NodeManager?

Reply:
Not strictly necessary, I debated with myself about this, but the idea was that the CLI commands would no nothing of the backend being used. If there is no node manager or something of the like, the CLI will have to initialize an particular implementation of the NodeStore. 

The other part of it was that not the CLI nor the Store would know about data validation. The NodeStore would just know how to store the data, and load data, and any data validation is responsibility of the NodeManager.

Reply:
The DefaultValueManager, might be a better example of any value this abstraction might offer